packages/run-protocol/src/feeDistributor.js

```diff
@@ -0,0 +1,308 @@
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { observeNotifier } from '@agoric/notifier';
+import { makeScalarSetStore } from '@agoric/store';
+
+/**
+ * @typedef {object} FeeCollector
+ *
+ * @property {() => ERef<Payment>} collectFees
+ */
+
+/**
+ * @typedef {object} PeriodicFeeCollector
+ * @property {() => FeeCollector} getCollector
+ * @property {() => Promise<void>} collectAndDistributeNow
+ * @property {() => void} stop
+ */
+
+/**
+ * @typedef {object} CollectibleContractFacet
+ * @property {() => Promise<Invitation>} makeCollectFeesInvitation
+ */
+
+/**
+ * wrapper to take the vaultFactory or AMM's creatorFacet, and make a function that
+ * will request an invitation and return a promise for a payment.
+ *
+ * @param {ERef<ZoeService>} zoe
+ * @param {ERef<CollectibleContractFacet>} creatorFacet
+ */
+export function makeContractFeeCollector(zoe, creatorFacet) {
```

Comment:
not imported. consider not exporting. TMK, contracts are to export only `start`.

also shouldn't this be `const makeContractFeeCollector = (zoe, creatorFacet) => {` ?

Reply:
I'm exporting for future tests.  I just never got to them.

> TMK, contracts are to export only `start`.

Not necessarily, as long as there is motivation.